File: server/hotrod/src/main/scala/org/infinispan/server/hotrod/event/KeyValueWithPreviousEventConverterExternalizer.scala
package org.infinispan.server.hotrod.event

import java.io.{ObjectInput, ObjectOutput}

import org.infinispan.commons.marshall.AbstractExternalizer

import scala.collection.JavaConversions._

/**
 * Externalizer for KeyValueWithPreviousEventConverter
 *
 * @author gustavonalle
 * @since 7.2
 */
class KeyValueWithPreviousEventConverterExternalizer[K,V] extends AbstractExternalizer[KeyValueWithPreviousEventConverter[K,V]] {
   def getTypeClasses = setAsJavaSet(Set(classOf[KeyValueWithPreviousEventConverter[K,V]]))

   override def readObject(input: ObjectInput) = new KeyValueWithPreviousEventConverter[K,V]()

   override def writeObject(output: ObjectOutput, `object`: KeyValueWithPreviousEventConverter[K,V]): Unit = {}
}

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/event/KeyValueWithPreviousEventConverterExternalizer.java
package org.infinispan.server.hotrod.event;

import java.io.IOException;
import java.io.ObjectInput;
import java.io.ObjectOutput;
import java.util.Set;

import org.infinispan.commons.marshall.AbstractExternalizer;
import org.infinispan.commons.util.Util;

/**
 * Externalizer for KeyValueWithPreviousEventConverter
 *
 * @author gustavonalle
 * @since 7.2
 */
public class KeyValueWithPreviousEventConverterExternalizer extends AbstractExternalizer<KeyValueWithPreviousEventConverter> {
   @Override
   public Set<Class<? extends KeyValueWithPreviousEventConverter>> getTypeClasses() {
      return Util.asSet(KeyValueWithPreviousEventConverter.class);
   }

   @Override
   public void writeObject(ObjectOutput output, KeyValueWithPreviousEventConverter object) throws IOException {
   }

   @Override
   public KeyValueWithPreviousEventConverter readObject(ObjectInput input) throws IOException, ClassNotFoundException {
      return new KeyValueWithPreviousEventConverter();
   }
}
